lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -89,18 +96,27 @@ export class TokenCache implements ITokenCache {
     /**
      * Helper function to load id tokens to msal-browser cache
      * @param idToken
-     * @param homeAccountId
+     * @param clientInfo
      * @param environment
      * @param tenantId
      * @param options
+     * @param authorityType
+     * @param homeAccountId
+     * @returns `CacheRecord` containing the id and account information that was loaded.
      */
-    private loadIdToken(idToken: string, homeAccountId: string, environment: string, tenantId: string, options: LoadTokenOptions): void {
+    private loadIdToken(idToken: string, clientInfo: string, environment: string, tenantId: string, options: LoadTokenOptions, authorityType?: AuthorityType, homeAccountId?: string): CacheRecord {
 
-        const idTokenEntity = IdTokenEntity.createIdTokenEntity(homeAccountId, environment, idToken, this.config.auth.clientId, tenantId);
         const idAuthToken = new AuthToken(idToken, this.cryptoObj);
+        const idTokenHomeAccountId = homeAccountId ?
+            homeAccountId :
+            authorityType !== undefined ?
+                AccountEntity.generateHomeAccountId(clientInfo, authorityType, this.logger, this.cryptoObj, idAuthToken) :
+                clientInfo;
```

Comment:
Can this be done without nesting ternaries?

Reply:
(I thought we had a lint rule to explicitly disallow this, maybe not)

Reply:
Sure, I can write it in a different way.